### PR TITLE
Deconv using CalcFactor dispatch

### PR DESCRIPTION
**Pull request by Affie**

### CODE REVIEW

File: src/services/NumericalCalculations.jl [hunk 1]
```diff
@@ -177,6 +167,105 @@ function _solveLambdaNumericMeas(
...
+
+# for deconv with the measurement a tangent vector, can dispatch for other measurement types.
+function (hypoCalcFactor::CalcFactorNormSq)(::Type{CalcDeconv}, M::AbstractManifold, Xc::AbstractVector)
+  ϵ = getPointIdentity(M)
+  X = get_vector(M, ϵ, Xc, DefaultOrthogonalBasis())
+  return hypoCalcFactor(CalcDeconv, X)
+end
+
+#NOTE Optim.jl version that assumes measurement is on the tangent
+function _solveLambdaNumericMeas_v2(
```
Comment (dehann) [inline src/services/NumericalCalculations.jl:186]:
```suggestion
# TODO test / dev for n-ary factor deconv
function _solveLambdaNumericMeas_v2(
```

File: src/services/NumericalCalculations.jl [hunk 2]
```diff
@@ -177,6 +167,105 @@ function _solveLambdaNumericMeas(
...
+end
+
+function approxDeconv_v2(
+  fcto::DFGFactor,
+  ccw::CommonConvWrapper = _getCCW(fcto);
+  N::Int = 100,
+  measurement::AbstractVector = sampleFactor(ccw, N),
+)
+  # NOTE comments kept from original...
+  # FIXME needs xDim for all variables at once? xDim = 0 likely to break?
```
Comment (Affie) [inline src/services/NumericalCalculations.jl:219]: TBD deprecate use of xDim

File: src/services/NumericalCalculations.jl [hunk 3]
```diff
@@ -177,6 +167,105 @@ function _solveLambdaNumericMeas(
...
+
+  # but what if this is a partial factor -- is that important for general cases in deconv?
+  _setCCWDecisionDimsConv!(ccw, 0) # ccwl.xDim used to hold the last forward solve getDimension(getVariableType(Xi[sfidx]))
```
Comment (Affie) [inline src/services/NumericalCalculations.jl:222]: remove xDim comment

File: src/services/NumericalCalculations.jl [hunk 4]
```diff
@@ -177,6 +167,105 @@ function _solveLambdaNumericMeas(
...
+
+  # FIXME This does not incorporate multihypo??
```
Comment (Affie) [inline src/services/NumericalCalculations.jl:224]:
```suggestion
  # FIXME This does not incorporate multihypo, Apply hyprecipe to full variable order list. But remember hyporecipe assignment must be found (NPhard)
```

File: src/services/NumericalCalculations.jl [hunk 5]
```diff
@@ -397,7 +395,7 @@ function (cf::CalcFactorNormSq)(::Type{CalcConv}, x)
   return sum(x->x^2, res)
 end
 
-function _buildHypoCalcFactor(ccwl::CommonConvWrapper, smpid::Integer, _slack)
+function _buildHypoCalcFactor(ccwl::CommonConvWrapper, smpid::Integer, slack=nothing)
```
Comment (Affie) [inline src/services/NumericalCalculations.jl:398]:
```suggestion
function _buildHypoCalcFactor(ccwl::CommonConvWrapper, smpid::Integer, _slack=nothing)
```

**Pull request merged**